contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -361,23 +364,62 @@ contract TrueCreditAgency is UpgradeableClaimable {
 
     function poke(ITrueFiPool2 pool) public {
         uint256 bitMap = usedBucketsBitmap;
-        CreditScoreBucket[256] storage creditScoreBuckets = buckets[pool];
         uint256 timeNow = block.timestamp;
         uint256 partialRate = riskPremium.add(utilizationAdjustmentRate(pool));
 
+        for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
+            if (bitMap & 1 == 0) {
+                continue;
+            }
+
+            _pokeSingleBucket(pool, uint8(i), timeNow, partialRate);
```

Comment:
This uint8(i) cast reminds me: the bucket loop iteration variables should all be uint8 instead of uint16. I'll add a task for this.

Reply:
On that topic we can also set MAX_CREDIT_SCORE = type(uint8).max.

Reply:
@yuchenlintt Unfortunately it would overflow after the last iteration and before the final check

Reply:
Ah, you're right. My bad then.